frontend/app/stores/banner/snapshots.ts

```diff
@@ -0,0 +1,54 @@
+import isEqual from 'lodash/isEqual'
+import { snapshot } from 'valtio'
+import { createDefaultBannerProfile } from '@shared/default-data'
+import { PROFILE_IDS, DEFAULT_PROFILE_NAME } from '@shared/types'
+import type { ProfileId, BannerProfile, Configuration } from '@shared/types'
+import { banner } from './store'
+
+const SNAP_STORAGE_KEY = 'wmt-banner-snapshots'
+
+const snapshots = new Map<ProfileId, BannerProfile>(
+  PROFILE_IDS.map((id, idx) => [
+    id,
+    createDefaultBannerProfile(DEFAULT_PROFILE_NAME[idx])
+  ])
+)
+
+export function setSnapshots(config: Configuration<'banner'>) {
+  Object.entries(config).forEach(([profileId, profile]) => {
+    snapshots.set(profileId as ProfileId, profile)
+  })
+
+  localStorage.setItem(SNAP_STORAGE_KEY, JSON.stringify(config))
+}
+
+export function isProfileDirty(id: ProfileId) {
+  const profile = snapshots.get(id)
+  if (!profile) {
+    return false
+  }
+
+  const snap = snapshot(banner.profiles[id])
+  return !isEqual(snap, profile)
```

Comment:
If we make use of `$modifiedAt`, we can avoid a relatively more expensive `isEqual` all the time.